### DISCUSSION TITLE
Banning someone who is not in your server

### DISCUSSION THREAD

**Discussion by CristianEduardMihai** [Rapptz/discord.py#6891] [Q&A] [answered]
Hi ! I wanna ban someone who is not in my server using his id. Found this old issue doing basically the same thing. 
Could someone help me do it in the updated version of discord.py ?
https://github.com/Rapptz/discord.py/issues/1223
This is the code I tried...but is outdated.

```
@bot.command(description="Ban")
@commands.has_permissions(administrator=True)
async def blacklist(ctx):
    guild = bot.guild
    await guild.ban(discord.Object(id=member_id))
```

**Comment by esmaycat** [comment, marked as answer]
`member_id` wouldn't be defined here. You could make ot an argument to the command and typehint it to an `int`? Also, I don't think that `bot.guild` exists - you probably want `ctx.guild` to get the current guild.

**Reply by CristianEduardMihai** [reply]
Did the second part. But I didn't understand the typehint part..
```
async def blacklist(ctx):
    guild = ctx.guild
    await guild.ban(discord.Object(id=member.id))
    await asyncio.sleep(1)
    await ctx.send("Done")
```

**Reply by esmaycat** [reply]
There is documentation on this [here](https://discordpy.readthedocs.io/en/latest/ext/commands/commands.html#basic-converters)

**Comment by CristianEduardMihai** [comment]
SOLVED IT. Here is the code in someone wants it:
```
@bot.command(description="Ban someone who is not on the server")  #blacklisting someone from joining
@commands.has_permissions(ban_members=True) #setting the permission
async def blacklist(ctx, user: discord.User):   #defining ctx and user
    guild = ctx.guild     #defining guild
    await guild.ban(discord.Object(id=user.id))  #banning the user
    await asyncio.sleep(1)   #wait 
    await ctx.send("Done") #say done
```
